src/Plugin.php

```diff
@@ -291,7 +291,8 @@ public static function enqueueGaDataLayerScripts() {
 
     wp_enqueue_script($handle . '_common', "$scripts/common.js", ['jquery'], FALSE, TRUE);
     wp_localize_script($handle . '_common', Plugin::PREFIX . '_settings', [
-      'datalayer_console_log' => (int) get_option('shop_analytics_datalayer_logging') ? 'on' : 'off'
+      'tc_enabled' => (bool) get_option('shop_analytics_tc_enabled'),
```

Comment:
Did you check which values are stored in the DB? The values in the DB are always stored as string so this could lead to unexpected results. Additionally if "on" and "off" are stored, the `(bool)` typecast will always be true as the string isn't empty in either case.